### PR TITLE
DOC: Clarify when csv separator is being parsed as regex. Resolves #10208

### LINKED ISSUE #10208: Regex C Engine Warning
Using `pd.read_csv(..., sep=", ", ...)` I'm now getting a warning about falling back to the C engine because regex parsing isn't supported in the C engine. That's fine, but this isn't actually using regex.

I don't have an idea for a good transition strategy, and maybe the ship has sailed, but perhaps there should be a separate `read_regex` or a `regex` keyword instead of emitting this warning for any string greater than length 1.

Pandas 0.16.0.

### PR DESCRIPTION
**Pull request by dukebody**
- [x] closes #10208
- [ ] tests added / passed -- only docs, not needed
- [x] passes `git diff upstream/master | flake8 --diff`
- [ ] whatsnew entry -- neccessary for docs?

### CODE REVIEW

File: pandas/io/parsers.py [hunk 1]
```diff
@@ -221,8 +221,9 @@ class ParserWarning(Warning):
 
 _sep_doc = """sep : str, default {default}
     Delimiter to use. If sep is None, will try to automatically determine
-    this. Regular expressions are accepted and will force use of the python
-    parsing engine and will ignore quotes in the data."""
+    this. Separators longer than 1 character will be interpreted as regular
+    expressions, will force use of the python parsing engine and will ignore
```
Comment (jreback) [inline pandas/io/parsers.py]: maybe show an example, as ``sep='\s+'`` DOES use the c-engine (its a bit of a special case), an example would be ``sep=`\r\t``` or somesuch

**Comment by jreback** [comment]
also update the docs in ``io.rst`` (with the same), though you can be

**Comment by dukebody** [comment]
Updated!

**Comment by jreback** [comment]
thanks!

**Pull request closed without merging**